CoroutineTemplate/domain/src/test/java/co/nimblehq/coroutine/domain/usecase/GetUsersUseCaseTest.kt

```diff
@@ -0,0 +1,54 @@
+package co.nimblehq.coroutine.domain.usecase
+
+import co.nimblehq.coroutine.domain.model.User
+import co.nimblehq.coroutine.domain.repository.UserRepository
+import io.kotest.matchers.shouldBe
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class GetUsersUseCaseTest {
+
+    private lateinit var mockRepository: UserRepository
+    private lateinit var usecase: GetUsersUseCase
+
+    private val user = User(
+        id = 1,
+        name = "name",
+        username = "username",
+        email = "email",
+        address = null,
+        phone = "",
+        website = ""
+    )
+
+    @Before
+    fun setup() {
+        mockRepository = mockk()
+        usecase = GetUsersUseCase(mockRepository)
+    }
+
+    @Test
+    fun `When calling request successfully, it returns success response`() = runBlockingTest {
+        val expected = listOf(user)
+        coEvery { mockRepository.getUsers() } returns expected
+
+        usecase.execute().run {
+            (this as UseCaseResult.Success).data shouldBe expected
+        }
+    }
+
+    @Test
+    fun `When calling request failed, it returns wrapped error`() = runBlockingTest {
```

Comment:
Since `runBlockingTest` will be deprecated on the newer version, should we use `runTest` instead? It would reduce the effort when we upgraded to a newer version. Migration guideline [here](https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-test/MIGRATION.md) for Kotlin version 1.6.0

Reply:
@hoangnguyen92dn that's correct but it's not in this PR scope tbh. We have another task to upgrade Gradle is https://github.com/nimblehq/android-templates/issues/163, @Tuubz we could consider upgrading Kotlin version too 🤓

Reply:
Yeah, upgrading Kotlin is a good idea. I'll create a new issue for it soon 🚀